app/assets/src/components/views/amr_heatmap/AMRHeatmapVis.jsx

```diff
@@ -0,0 +1,193 @@
+import React from "react";
+import PropTypes from "prop-types";
+
+import Heatmap from "~/components/visualizations/heatmap/Heatmap";
+import { getAMRCounts } from "~/api/amr";
+
+import cs from "./amr_heatmap_vis.scss";
+
+const METRIC_ALLELES = "alleles";
+const METRIC_GENES = "genes";
+
+export default class AMRHeatmapVis extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      viewLevel: METRIC_GENES,
+      metric: "coverage",
+    };
+
+    this.heatmap = null;
+  }
+
+  componentDidMount() {
+    this.requestAMRCountsData(this.props.sampleIds);
+  }
+
+  componentDidUpdate() {
+    this.renderHeatmap();
+  }
+
+  async requestAMRCountsData(sampleIds) {
+    const rawSampleData = await getAMRCounts(sampleIds);
+    const samplesWithAMRCounts = rawSampleData.filter(
+      sampleData => sampleData.error === ""
+    );
+    const [samples, genes, alleles] = this.createLabels(samplesWithAMRCounts);
+    this.setState({
+      samplesWithAMRCounts,
+      samples,
+      genes,
+      alleles,
+    });
+  }
+
+  createLabels(sampleData) {
+    const sampleLabels = [];
+    const genes = {};
+    const alleles = {};
+    sampleData.forEach(sample => {
+      sampleLabels.push({ label: sample.sample_name });
+      sample.amr_counts.forEach(amrCount => {
+        genes[amrCount.gene] = true;
+        alleles[amrCount.allele] = true;
+      });
+    });
+    const geneLabels = Object.keys(genes).map(gene => {
+      return { label: gene };
+    });
+    const alleleLabels = Object.keys(alleles).map(allele => {
+      return { label: allele };
+    });
+
+    return [sampleLabels, geneLabels, alleleLabels];
+  }
+
+  //*** Following functions must be called after the component has updated ***
+  //*** (i.e. after the component has requested AMR data and updated state) ***
+
+  createHeatmapLabels() {
+    const viewLevel = this.state.viewLevel;
+    switch (viewLevel) {
+      case METRIC_ALLELES: {
+        return this.state.alleles;
+      }
+      case METRIC_GENES: {
+        return this.state.genes;
+      }
+    }
+  }
+
+  assembleAlleleValues(alleleLabels) {
+    const sampleData = this.state.samplesWithAMRCounts;
+    const alleleValues = {
```

Comment:
Also, it seems a little weird to assemble both depth and coverage values, and then throw one of them away in `computeHeatmapValues`. Could you pass whether you want "depth" or "coverage" into the function, and then just gather that metric?